src/js/stores/middlewares/messages.js

```diff
@@ -0,0 +1,24 @@
+import { ADD_MESSAGE } from 'actions/conversation-actions';
+import { showSettingsNotification } from 'actions/app-state-actions';
+
+
+export function firstMessage(store) {
+    return function(next) {
+        return function(action) {
+            let result = next(action);
+            let state = store.getState();
+            if (state.appState.settingsEnabled && action.type === ADD_MESSAGE && action.message.role === 'appUser' && !state.user.email) {
```

Comment:
I guess the reducer would need to do something with this `FIRST_USER_MESSAGE ` event, not the view itself. Does it make sense to check state and dispatch `showSettingsNotification()` from a reducer?